### PR TITLE
auto device setup

### PR DESCRIPTION
**Pull request by acheong08**
For the lazy folks.

Adds
```
  -auto-setup
    	Auto setup TUN device (Linux only)
```
to the command line options.

### CODE REVIEW

**Comment by acheong08** [comment]
Not sure how to do it for Windows/MacOS

**Comment by xjasonlyu** [comment]
It's an excellent feature! I wanted to do it long time ago. Thanks 😊

**Comment by acheong08** [comment]
I'm a bit confused by the windows example (since I don't use it)

What is `<IF NUM>` in ```
route add 0.0.0.0 mask 0.0.0.0 192.168.123.1 if <IF NUM> metric 5
```?

**Comment by acheong08** [comment]
Don't have a Windows or MacOS device. Requires testing.

File: main.go [hunk 1]
```diff
@@ -21,8 +24,12 @@ var (
 
 	configFile  string
 	versionFlag bool
+
+	autoSetup bool
 )
 
+const privateIP = "10.10.10.10"
```
Comment (xjasonlyu) [inline main.go:31]: Hardcoded IP is not recommended. Sometimes it may cause conflicts.

File: main.go [hunk 2]
```diff
@@ -61,12 +69,104 @@ func main() {
 		}
 	}
 
+	if autoSetup && runtime.GOOS == "linux" {
```
Comment (xjasonlyu) [inline main.go:72]: We can use build tags here instead of checking it at the runtime.

**Review by xjasonlyu — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (acheong08) [reply main.go]: [-private-ip option](https://github.com/xjasonlyu/tun2socks/pull/284/commits/b672a3e14789b0fa58b54ec6f7d205f6889b26a6)

File: main.go (hunk 2 above)
Comment (acheong08) [reply main.go:72]: How do I do this?

I understand you can do 
```go
// +build <GOOS>
```
But that seems to cover the entire file

**Pull request closed without merging**